MD Trajectory Analysis tools

chrisbarnettster opened the pull request:
Hi everyone 
some updates and new tools 
1.  Updated Ramachandran tool, selection done using a YAML syntax (mdanalysis/ramachandran_plots.*)
2.  New Ramachandran tool for proteins (mdanalysis/ramachandran_auto_protein.*)
3. End to End distance timeseries and histogram for proteins (mdanalysis/end-to-end.*)
4. Trajectory tool which allows for selections and multijoin. (mdtraj/traj_select_and_merge.xml)

bgruening commented on tools/mdanalysis/end-to-end.xml:
```diff
@@ -0,0 +1,86 @@
+<tool id="mdanalysis_endtoend" name="End-to-End Analysis" version="@VERSION@+galaxy@GALAXY_VERSION@">
+	<description>- End-to-End distance timeseries and histogram for the given selections</description>
+	<macros>
+		<import>macros.xml</import>
```
indentation should be 4spaces and not tab

bgruening commented on tools/mdanalysis/end-to-end.xml:
```diff
@@ -0,0 +1,86 @@
+	</macros>
+	<expand macro="requirements"/>
+	<command detect_errors="exit_code">
+		<![CDATA[
```
indentation ...

bgruening commented on tools/mdanalysis/ramachandran_auto_protein.xml:
```diff
@@ -0,0 +1,143 @@
+  </command>
+  <inputs>
+    <expand macro="analysis_inputs"/>
+    <param name="segid1" type="text" value="PROA" label="Segment ID of atom 1"/>
```
this needs a validator, I think we have in other tools already some which you can reuse.

bgruening commented on tools/mdanalysis/ramachandran_auto_protein.xml:
```diff
@@ -0,0 +1,143 @@
+      </param>
+      <when value="No"></when>
+      <when value="Yes">
+        <param name="resname" type="text" value="ALA" label="Resname"/>
```
can this be optional=True and then the conditional can be removed?

chrisbarnettster commented:
OK I think I got this sorted. Removed the conditional. Cleaned up those awful tabs. Included validators. Any further comments @tsenapathi @bgruening ?

simonbray commented on tools/mdanalysis/end-to-end.xml:
```diff
@@ -0,0 +1,91 @@
+    <description>- End-to-End distance timeseries and histogram for the given selections</description>
+    <macros>
+        <import>macros.xml</import>
+        <token name="@GALAXY_VERSION@">0</token>
```
Nice :+1:

simonbray commented on tools/mdanalysis/end-to-end.xml:
```diff
@@ -0,0 +1,91 @@
+]]></command>
+    <inputs>
+        <expand macro="analysis_inputs"/>
+        <param name="segid1" type="text" value="PROA" label="Segment ID of atom 1">
```
can you add some help text to these params?

simonbray commented on tools/mdanalysis/ramachandran_auto_protein.xml:
```diff
@@ -0,0 +1,127 @@
+]]></command>
+    <inputs>
+        <expand macro="analysis_inputs" />
+        <param name="segid1" type="text" value="PROA" label="Segment ID of atom 1">
```
why atom 1? should it be 'of the protein'?

chrisbarnettster commented:
@simonbray Thanks updates made.

chrisbarnettster commented:
@simonbray @bgruening could you squash and merge?

bgruening commented:
Of course!
